apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1.Tests/SubscriberClientTest.cs

```diff
@@ -884,20 +884,23 @@ public void LeaseExtension()
                     await fake.TaskHelper.ConfigureAwait(doneTask);
                     Assert.Equal(1, fake.Subscribers.Count);
                     DateTime S(int seconds) => fake.Time0 + TimeSpan.FromSeconds(seconds);
-                    Assert.Equal(new[] { S(0), S(5), S(20), S(25), S(40), S(45), S(60), S(65) }, fake.Subscribers[0].Extends.Select(x => x.Time));
+                    var expectedExtends =
+                        isExactlyOnceDelivery ? new[] { S(0), S(5), S(10), S(15), S(20), S(25), S(30), S(35), S(40), S(45), S(50), S(55), S(60), S(65) }
```

Comment:
I think I'd prefer this test to be split in two distinct tests. Theories are not a way to reuse setup code, theories are a way to prove to a whole set of inputs (a theory) behave in the same way.

And the same for the rest of the modifications below.

Reply:
Now the test is common for eod and regular subs.